src/ImageCache.cpp

```diff
@@ -55,6 +56,9 @@ ImageCache *IMAGECACHE; // global and accessible from anywhere in our program
 static std::map<RString, RageSurface*> g_ImagePathToImage;
 static int g_iDemandRefcount = 0;
 
+/* Synchronizes access to g_ImagePathToImage and ImageCache::ImageData. */
+static RageMutex g_ImageCacheMutex("ImageCache");
```

Comment:
I wonder how much contention we'll have with this lock since I imagine loading images from disk could be heavy. I think this is totally fine for now, but something we can see if we can further optimize (only really if necessary though).